test/tools/Modules/HttpListener/HttpListener.psm1

```diff
@@ -214,19 +214,32 @@ Function Start-HTTPListener {
                     $response.Close()
                 }
             }
+            catch
+            {
+                $errormsg = $_ | convertto-json
+                Write-Error $errormsg
+            }
             finally
             {
                 $listener.Stop()
+                Write-Information "Listener is stopped" -InformationAction Continue
             }
         }
 
-        if ($AsJob)
+        if ($Foreground)
         {
-            Start-Job -ScriptBlock $script -ArgumentList $Port
+            & $script -Port $Port
         }
         else
         {
-            & $script -Port $Port
+            $ps = [powershell]::Create()
+            $null = $ps.AddScript($script)
+            $null = $ps.AddParameter("port",$port)
+            $AsyncResponse = $ps.BeginInvoke()
+            [pscustomobject]@{
+                PowerShell = $ps
+                AsyncResponse  = $AsyncResponse
```

Comment:
the async response can be used for debugging purposes. It's better to keep track of it.